### ISSUE TITLE
Warning when hardlinking and metadata tagging enabled

### ISSUE THREAD

**Issue by tommyalatalo** [lidarr/Lidarr#1016]
**Is your feature request related to a problem? Please describe.**
When a torrent client is used, along with hardlinking and metadata tagging of the files, the files in the torrent client are corrupted.

**Describe the solution you'd like**
There should be implemented a prominent warning that if there is a torrent client used with both hardlinking and metadata tagging enabled, the torrent client data is very likely to be corrupted.

**Comment by MammothJerk** [comment]
I was wondering why only my music torrents were getting these errors, this seems to be my problem as well and it was driving me crazy.

**Comment by IllMethods** [comment]
Yep, having the same issues...

Mine setup

Transmission
Lidarr
Hardlinking
Rescan and Fingerprinting set to always

Not sure what else to add, but only happens to torrents involved with Lidarr.

**Comment by IllMethods** [comment]
Just found this...

Overview
Lidarr can write metadata for matched tracks to the audio file (e.g. ID3 tags). This data contains the track title, artist track number etc. as well as MusicBrainz identifiers. The result should be similar to that of MusicBrainz Picard, though we currently don't populate all of the fields that Picard does.

Note that if you enable this option then the files on disk will be changed. If the files are part of a torrent you are seeding then the torrent will be corrupted.

**Comment by Elmeromoto** [comment]
Yeah this is a horrible setting.  It can (and does!) completely destroy a person's ratio and break their seeding. It needs a lot more controls built around it.

**Comment by john-parton** [comment]
If hardlinking used "reflink" or "block cloning" on linux, this wouldn't be a problem. Is there a way to tell Lidarr to use `cp --reflink`?

**Comment by mat926** [comment]
I've been having this problem too. The biggest downside is I have to copy the files and that would take up twice the disk space.  Is there a better option available that doesn't take up twice the space and still have metadata tagging enabled?

**Comment by john-parton** [comment]
If your filesystem supports it (btrfs and recent zfs), cp --reflink always will make a COW copy of the file. This original copy wouldn't take up any space, but once the file is modified, it will take up some. The amount is dependent on filesystem implementation and configuration, as well as dependent on the exact operation of the tagging code.

Most likely the benefit would be largely negated for small files like audio files, but it could be nice to test.

**Comment by Lord-Valen** [comment]
> If hardlinking used "reflink" or "block cloning" on linux, this wouldn't be a problem. Is there a way to tell Lidarr to use `cp --reflink`?

https://github.com/Lidarr/Lidarr/blob/d8850af0193846df6de6ec38d67bfabffe58e86b/src/NzbDrone.Common/Disk/DiskTransferService.cs#L346-L391

Seems that lidarr will try to reflink as long as the source and target disks use the same format out of btrfs, zfs, and xfs, and it is not told to hardlink. "Copy" will reflink and keep the original and copy.  "Move" will reflink and delete the original afterward. It wouldn't be too hard to have the same flow in the hardlink mode, but seeing this it seems to be a deliberate choice. If we want reflinks, we should probably just use the copy mode.
